README.md

```diff
@@ -80,50 +113,136 @@ Options:
+```
+┌─────────────┐                        ┌─────────────┐
+│  Postgres   │                        │  Snowflake  │
+├─────────────┤                        ├─────────────┤
+│             │                        │             │
+│             │                        │             │
+│             │  ┌─────────────┐       │ table with  │
+│ table with  ├──┤ replication ├──────▶│ ?maybe? all │
+│lots of rows!│  └─────────────┘       │  the same   │
+│             │                        │    rows.    │
+│             │                        │             │
+│             │                        │             │
+│             │                        │             │
+└─────────────┘                        └─────────────┘
+```
 
-The algorithm goes like this:
+In order to check whether the two tables are the same, **data-diff** splits
+the table into `--bisection-factor=10` segments.
 
-0. Table segments `A` and `B` are set to the two tables for comparison.
+We also have to choose which columns we want to checksum. In our case, we care
+about `id` and `updated_at`. `updated_at` is updated every time the row is, and
+we have an index on it.
 
-1. Calculate the checksums on `A` and `B` using MD5.
+**data-diff** starts by querying both databases for the `min(id)` and `max(id)`
+of the table. Then it splits the table into `--bisection-factor=10` segments of
+`1M/10 = 100K` keys each:
 
-    1. If they are the same, the tables are considered equal. Stop.
+```
+┌──────────────────────┐              ┌──────────────────────┐
+│       Postgres       │              │      Snowflake       │
+├──────────────────────┤              ├──────────────────────┤
+│      id=1..100k      │              │      id=1..100k      │
+├──────────────────────┤              ├──────────────────────┤
+│    id=100k..200k     │              │    id=100k..200k     │
+├──────────────────────┤              ├──────────────────────┤
+│    id=200k..300k     ├─────────────▶│    id=200k..300k     │
+├──────────────────────┤              ├──────────────────────┤
+│    id=300k..400k     │              │    id=300k..400k     │
+├──────────────────────┤              ├──────────────────────┤
+│         ...          │              │         ...          │
+├──────────────────────┤              ├──────────────────────┤
+│      900k..100k      │              │      900k..100k      │
+└───────────────────▲──┘              └▲─────────────────────┘
+                    ┃                  ┃
+                    ┃                  ┃
+                    ┃ checksum queries ┃
+                    ┃                  ┃
+                  ┌─┻──────────────────┻────┐
+                  │        data-diff        │
+                  └─────────────────────────┘
+```
 
-    2. If their size is below the threshold, diff them locally and print the results.
+Now **data-diff** will start running `--threads=1` queries in parallel that
```

Comment:
Don't forget to add `--threads` to the options reference up above in the "how to use" section.

Reply:
Yep, we're just revising the wording for this in another PR — where we'll do this 👍🏻 

We want `--threads=1` to mean "1 simultaneous query per database", today it doesn't really mean that...